target-app/Home/Views/HomeViewController.swift

```diff
@@ -137,7 +160,37 @@ extension HomeViewController: BottomSheetPresenter {
     }
 }
 
-extension HomeViewController {
+extension HomeViewController: MKMapViewDelegate{
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        guard annotation is MKPointAnnotation else { print("no mkpointannotaions"); return nil }
+        
+        let reuseId = "pin"
+        var pinView = mapView.dequeueReusableAnnotationView(withIdentifier: reuseId) as? MKPinAnnotationView
+        
+        if pinView == nil {
+            pinView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
+            pinView!.canShowCallout = true
+            pinView!.rightCalloutAccessoryView = UIButton(type: .infoDark)
+            pinView!.pinTintColor = UIColor.black
+        }
+        else {
+            pinView!.annotation = annotation
+        }
+        return pinView
+    }
+    
+    func mapView(_ mapView: MKMapView, didSelect view: MKAnnotationView) {
+        //TODO: tapped on pin
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        if control == view.rightCalloutAccessoryView {
+            if (view.annotation?.title!) != nil {
+                //TODO: add tap action
+            }
+        }
```

Comment:
```suggestion
        if control == view.rightCalloutAccessoryView && view.annotation?.title != nil {
          //TODO: add tap action
        }
```
Everytime you can, ps void force unwrap